src/modules/search/components/Search.tsx

```diff
@@ -64,21 +61,20 @@ export function Search({ isOnBackground }: { isOnBackground?: boolean }) {
     setSearchTerm("");
   }, [category]);
 
-  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
   useEffect(() => {
-    router.replace({ query: { q: searchTerm } });
+    setAppState({ search: searchTerm }, { type: "replace" });
```

Comment:
Idea (non-blocking): In theory there could be `replaceAppState` and `pushAppState` to mimick what other routing APIs typically do, but this API is absolutely fine for now, too :)

Reply:
Yeah, I thought about that too. My intention was to decouple the concept of state from routing and query parameters with this API. Initially, I wanted to just allow push only, so people can always go back using the browser history as this is a good practice. I quickly realized, this will not work for the map, so I allowed the usage of replace too, but I decided to hide it a bit behind an option to discourage its usage. No hard feelings here, I can modify it to match the routers API more closely. What do you think?